notification/src/handler/notification/notification.handler.js

```diff
@@ -83,23 +108,33 @@ async function updatePaymentWithRepeater(
       )
       break
     } catch (err) {
-      if (err.body.statusCode !== 409)
-        throw new Error(
-          `Unexpected error during updating a payment with ID: ${currentPayment.id}. Exiting. ` +
-            `Error: ${JSON.stringify(serializeError(err))}`
-        )
+      if (err.statusCode !== 409)
+        throw new CommercetoolsError({
+          stack: err.stack,
+          message:
+            `Unexpected error on payment update with ID: ${currentPayment.id}.` +
+            `Failed actions: ${JSON.stringify(
+              _obfuscateNotificationInfoFromActionFields(updateActions)
+            )}`,
+          statusCode: err.statusCode,
+        })
       retryCount += 1
       if (retryCount > maxRetry) {
         retryMessage =
           'Got a concurrent modification error' +
           ` when updating payment with id "${currentPayment.id}".` +
           ` Version tried "${currentVersion}",` +
           ` currentVersion: "${err.body.errors[0].currentVersion}".`
-        throw new Error(
-          `${retryMessage} Won't retry again` +
+        throw new CommercetoolsError({
+          stack: err.stack,
+          message:
+            `${retryMessage} Won't retry again` +
```

Comment:
I noticed above that on retry we do recalculate update actions but we do not really check if newly update actions would be empty or not?

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/629/commits/698ae1e0492ff9c74e1cb6cde34b7d16a29647ef

Reply:
looks like a bug, good catch: https://github.com/commercetools/commercetools-adyen-integration/pull/629/commits/0d3ed73a826676ff40857781c518026c3ffd74d6